rubin_scheduler/scheduler/basis_functions/mask_basis_funcs.py

```diff
@@ -269,37 +281,35 @@ def _calc_value(self, conditions, indx=None):
             out_of_bounds = np.where((future_az - self.min_az) % (two_pi) > az_range)[0]
             result[out_of_bounds] = np.nan
         # Check the conditions objects azimuth limits, now and future
-        if (conditions.tel_az_limits is not None) and (len(conditions.tel_az_limits) > 0):
+        if (conditions.sky_az_limits is not None) and (len(conditions.sky_az_limits) > 0):
             combined = np.zeros(hp.nside2npix(self.nside), dtype=float)
-            for limits in conditions.tel_az_limits:
+            for limits in conditions.sky_az_limits:
                 in_bounds = np.ones(hp.nside2npix(self.nside), dtype=float)
                 min_az = limits[0]
                 max_az = limits[1]
                 if np.abs(max_az - min_az) < two_pi:
-                    az_range = (max_az - min_az) % (two_pi) - 2 * conditions.altaz_limit_pad
+                    az_range = (max_az - min_az) % (two_pi) - 2 * self.altaz_limit_pad
```

Comment:
Hmm. What we actually need is to move the edges of the healpix maps "inward" away from the edges by a fixed amount of on-sky distance .. 
I wonder if another answer to this would be for the on-sky tesselation to handle the healpix -> field center mappings and only choose field centers which correspond to areas which are completely covered by valid healpix values. (that way, it's not even just the altaz shadow mask which benefits, but also the moon avoidance zone or wind, etc.). 
I'll modify something here, but an option to consider.

Reply:
I had a go at various versions of this. 
An accurate way to go might be to do something like 
```
indx = np.arange(0, hp.nside2npix(nside))
disc_region = np.ones(hp.nside2npix(nside))
for ipix in indx:
    vec = hp.pix2vec(nside=nside, ipix=ipix)
    ipix_disc = hp.query_disc(nside=nside, vec=vec, radius=np.radians(2))
    disc_region[ipix] = region[ipix_disc].max()
```
But this is fairly slow -- might be something to consider when looking at a combined reward function and then picking out a particular field pointing from the tesselation. 

Another option was to try to add a tophat kind of smoothing function to the healpix map, using healpy. However, this didn't work as well as I'd hoped -- even after defining the tophat beam function, you still have to do some manual choices for the cutoff value in the resulting smoothed reward function in order to find the mask edges. 

So, I looked at the difference between a simple gaussian beam smoothing with healpy vs. the query_disc approach above, and found an option that was still fairly fast and also fairly close to the result from query_disc (if slightly more conservative) and have implemented that instead. 

The time required for this basis function to calculate reward does still go from 1ms to ~4.5ms (just timing it with %%timeit in a notebook cell, and on my (old) laptop).